### ISSUE TITLE
CPM15 and CPM17 dataset

### ISSUE THREAD

**Issue by HaotianWang86** [vqdang/xy_net#5]
Could you please update the CPM15 and CPM17 dataset on Github?

**Comment by simongraham** [comment]
Hi Haotian. 

We are currently in the process of updating the repository. In the next couple of weeks, we will have everything updated. 

Simon

**Comment by zalach** [comment]
are the datasets available?

**Comment by simongraham** [comment]
Sorry for the delay with this- as you can see, we have had a major restructure of our recent work.

We have organised the CPM datasets in the below link:

https://drive.google.com/open?id=1l55cv3DuY-f7-JotDN7N5nbNnjbLWchK

The labels are in the same format that the organisers of the CPM competitions used. I.e the a single vector is given with the size of the vector denoting the number of pixels. The dimensions of the image are at the top are given at the top of the csv file. Simply read in the vector and reshape to the given dimension to get the labels.

**Comment by HaotianWang86** [comment]
Hi Simon,
Thanks for your CPM15, CPM17 dataset. The google drive link works for me. But CPM 15 dataset only have train dataset, CPM 17 has both train and test dataset. How to do you split the test data for CPM 15 dataset? Also, I have read your new hover-net paper. In the paper, it said the CoNSeP dataset will be released in 'https://warwick.ac.uk/fac/sci/dcs/research/tia/data/'. I have checked the website, but the dataset hasn't posted on it yet. Can I ask what's the more exact time will release the dataset on the website? In the next several weeks or may later?


Thanks
Haotian





------------------ 原始邮件 ------------------
发件人: "Simon Graham"<notifications@github.com>;
发送时间: 2019年7月4日(星期四) 晚上6:28
收件人: "vqdang/hover_net"<hover_net@noreply.github.com>;
抄送: "Max王昊天"<345650307@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [vqdang/hover_net] CPM15 and CPM17 dataset (#5)




Sorry for the delay with this- as you can see, we have had a major restructure of our recent work.
 
We have organised the CPM datasets in the below link:
 
https://drive.google.com/open?id=1l55cv3DuY-f7-JotDN7N5nbNnjbLWchK
 
The labels are in the same format that the organisers of the CPM competitions used. I.e the a single vector is given with the size of the vector denoting the number of pixels. The dimensions of the image are at the top are given at the top of the csv file. Simply read in the vector and reshape to the given dimension to get the labels.

**Comment by simongraham** [comment]
Hi @HaotianWang86 ,

Yes, we only used the CPM 15 train dataset within our experiments. This was used within the combined CPM validation set within our generalisation study. I.e, for this generalisation test we used the CPM17 train, CPM17 test and CPM15 train sets together, forming a dataset of 32+32+15 = 79 images. 

For the CoNSeP dataset, as displayed in our footnote we will release this dataset only when the paper is accepted for publication - this work is still a pre-print. 

Simon

**Comment by HaotianWang86** [comment]
I got it, Thank you Simon.


Haotian




------------------ 原始邮件 ------------------
发件人: "Simon Graham"<notifications@github.com>;
发送时间: 2019年7月6日(星期六) 晚上11:55
收件人: "vqdang/hover_net"<hover_net@noreply.github.com>;
抄送: "Max王昊天"<345650307@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [vqdang/hover_net] CPM15 and CPM17 dataset (#5)




Hi @HaotianWang86 ,
 
Yes, we only used the CPM 15 train dataset within our experiments. This was used within the combined CPM validation set within our generalisation study. I.e, for this generalisation test we used the CPM17 train, CPM17 test and CPM15 train sets together, forming a dataset of 32+32+15 = 79 images.
 
For the CoNSeP dataset, as displayed in our footnote we will release this dataset only when the paper is accepted for publication - this work is still a pre-print.
 
Simon

**Comment by HaotianWang86** [comment]
Yes, the dataset works for me.




------------------ 原始邮件 ------------------
发件人: "zalach"<notifications@github.com>;
发送时间: 2019年7月3日(星期三) 晚上6:45
收件人: "vqdang/hover_net"<hover_net@noreply.github.com>;
抄送: "Max王昊天"<345650307@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [vqdang/hover_net] CPM15 and CPM17 dataset (#5)




are the datasets available?

**Comment by simongraham** [comment]
@HaotianWang86 @zalach 

To update, we have re-organised the data in the below link to make it easier to use. In the below link, we have organised the CPM-15, CPM-17 and Kumar dataset. For each dataset, we provide Images, Labels and Overlay folders:

- Images: original RGB images
- Labels: numpy array of instance segmentation labels with the same size as original image. Each nuclear instance is labelled from 1 to n, where n is the number of nuclei.
- Overlay: Labels overlaid on top of the original images for visualisation.

Note, different colours of the nuclear boundaries denote different instances. However, we randomly allocate the colour and therefore we do not guarantee that neighbouring instances have contrasting colours. 

To load the labels, simply use `numpy.load(label_filename)`

https://drive.google.com/drive/folders/1l55cv3DuY-f7-JotDN7N5nbNnjbLWchK?usp=sharing

**Comment by simongraham** [comment]
To update, we now use `.mat` files for ground truth for consistency in the repo. Refer to #31 for more info.

**Comment by sennnnn** [comment]
Hi, @simongraham. 
Where is the raw CPM17?

**Comment by wzr0108** [comment]
Other papers say that the test set of CPM-17 has 8 images, but the test set I downloaded here has 32 images. which is correct?

**Comment by windygooo** [comment]
> 

Have the same question.

**Comment by windygooo** [comment]
> Other papers say that the test set of CPM-17 has 8 images, but the test set I downloaded here has 32 images. which is correct?

Have the same question.

**Comment by vadori** [comment]
> > Other papers say that the test set of CPM-17 has 8 images, but the test set I downloaded here has 32 images. which is correct?
> 
> Have the same question.

Hi! Did you find an answer to this? Thank you!

**Comment by qiucan1** [comment]
> @HaotianWang86 @zalach
> 
> 为了更新，我们重新组织了以下链接中的数据，使其更易于使用。在下面的链接中，我们组织了 CPM-15、CPM-17 和 Kumar 数据集。对于每个数据集，我们提供 Images、Labels 和 Overlay 文件夹：
> 
> * 图片：原始RGB图像
> * 标签：与原始图像大小相同的实例分割标签的 numpy 数组。每个核实例都标记为从 1 到 n，其中 n 是原子核的数量。
> * 叠加：标签叠加在原始图像之上，以便可视化。
> 
> 请注意，核边界的不同颜色表示不同的实例。但是，我们随机分配颜色，因此我们不能保证相邻实例具有对比色。
> 
> 要加载标签，只需使用`numpy.load(label_filename)`
> 
> https://drive.google.com/drive/folders/1l55cv3DuY-f7-JotDN7N5nbNnjbLWchK?usp=sharing

hello could you please tell me witch software to create the dataset?Is it matlab?

**Comment by JJ999888** [comment]
May I ask if the images and numbers of the four different tumor types in the CPM-17 dataset provided correspond to each other? For example, are image_00 to image_07 images of the first tumor type?
